AirCasting/CoreData/NSManagedObjectContext+Utils.swift

```diff
@@ -41,13 +41,14 @@ extension NSManagedObjectContext {
         throw MissingSessionEntityError(uuid: uuid)
     }
     
-    func optionalExistingSession(uuid: SessionUUID) throws -> SessionEntity? {
+    func optionalExistingSession(uuid: SessionUUID) -> SessionEntity? {
         do {
             return try existingSession(uuid: uuid)
         } catch is MissingSessionEntityError {
             return nil
         } catch {
-            throw error
+            Log.error("Error when saving changes in \(uuid) session: \(error.localizedDescription)")
+            return nil
         }
     }
```

Comment:
I've checked the call chain of the method and in the end the only error handling was this line of logging, so I moved it here and made the method not throw, to handle the nil cases conveniently from the client code